### PR TITLE
Add GitHub actions

### PR DESCRIPTION
**Pull request by Rashmil-1999**
I have created one Github Action file that runs check on the Unit Tests that has been written when a new pull requests is opened.

### CODE REVIEW

**Review by ka7eh — COMMENTED** [review]
Looks like this is forked from your previous branch, which comes with some of its changes, but not the latest ones you committed. For future reference and to make it easier to review, we should try to fork from `develop` for each issue, unless the issue depends on another open branch. In such cases, the target branch for the PR should be the original branch that is forked.

File: .github/workflows/pull_request_check.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Install python requirements
+        run: |
+          pip install -U pip
+          pip install poetry
+          poetry install --no-dev
+        env:
+          POETRY_VIRTUALENVS_CREATE: false
+
+      - name: Setup the database before running migrations
+          run: |
```
Comment (ka7eh) [inline .github/workflows/pull_request_check.yml:48]: There are some indention issues here, which makes the file invalid. You can see the error under the Action tab too: https://github.com/Oceans-1876/challenger-api/actions

One good way to test and debug actions locally is this package: https://github.com/nektos/act
It doesn't catch all the issues that might come up in an actual run, but can help with many others.

File: .github/workflows/pull_request_check.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+name: Perform unit test on the new pull requests.
+
+on:
```
Comment (ka7eh) [inline .github/workflows/pull_request_check.yml:3]: Let's also include `workflow_dispatch` trigger so we can run each action manually too.

File: .github/workflows/pull_request_check.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+jobs:
+  project-setup-and-run-tests:
+    runs-on: ubuntu-latest
+
+     # Service containers to run with `runner-job`
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres
```
Comment (ka7eh) [inline .github/workflows/pull_request_check.yml:16]: Does this image support `postgis`? We might have to use another image that comes with `postgis` installed.

**Comment by Rashmil-1999** [comment]
> Looks like this is forked from your previous branch, which comes with some of its changes, but not the latest ones you committed. For future reference and to make it easier to review, we should try to fork from `develop` for each issue, unless the issue depends on another open branch. In such cases, the target branch for the PR should be the original branch that is forked.

Yes, I know, I realised it soon when I made the PR, but since there was no significant feature present in this PR, I did not `rebase` the branch to develop and went ahead with it. I will keep this in mind for future feature integration.

File: .github/workflows/pull_request_check.yml (hunk 1 above)
Comment (Rashmil-1999) [reply .github/workflows/pull_request_check.yml:48]: Yeah, that was one thing I noticed, and was going to ask you in today's meet. Let me look into it and will try it out.

File: .github/workflows/pull_request_check.yml (hunk 3 above)
Comment (Rashmil-1999) [reply .github/workflows/pull_request_check.yml:16]: I'll search about this and do the needful.

File: .github/workflows/pull_request_check.yml (hunk 2 above)
Comment (Rashmil-1999) [reply .github/workflows/pull_request_check.yml:3]: Okay, I will add it.

File: .github/workflows/pull_request_check.yml (hunk 1 above)
Comment (Rashmil-1999) [reply .github/workflows/pull_request_check.yml]: I installed the `act`  package, but it seems to utilize a lot of your computer resources.

File: .github/workflows/pull_request_check.yml (hunk 2 above)
Comment (Rashmil-1999) [reply .github/workflows/pull_request_check.yml]: I have added the `workflow_dispatch` trigger too. However, in the Actions tab, as indicated in their documentation, I can't seem to find the `run Workflow` option even after adding the trigger.

Comment (Rashmil-1999) [reply .github/workflows/pull_request_check.yml]: I figured the issue with the `workflow_dispatch` event not showing the ability to run from the `Actions` tab. Apparently the workflow should be on the `default` branch for it to give us the ability to run from the `Actions` tab.

**Pull request merged**